File: Inxi.NET/Base/HardwareParserBase.vb

'    Inxi.NET  Copyright (C) 2020-2021  EoflaOE
'
'    This file is part of Inxi.NET
'
'    Inxi.NET is free software: you can redistribute it and/or modify
'    it under the terms of the GNU General Public License as published by
'    the Free Software Foundation, either version 3 of the License, or
'    (at your option) any later version.
'
'    Inxi.NET is distributed in the hope that it will be useful,
'    but WITHOUT ANY WARRANTY; without even the implied warranty of
'    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
'    GNU General Public License for more details.
'
'    You should have received a copy of the GNU General Public License
'    along with this program.  If not, see <https://www.gnu.org/licenses/>.

Imports System.Management
Imports Claunia.PropertyList
Imports Newtonsoft.Json.Linq

Public MustInherit Class HardwareParserBase
    Implements IHardwareParser

    Public Overridable Function Parse(InxiToken As JToken, SystemProfilerToken As NSArray) As HardwareBase Implements IHardwareParser.Parse
        Throw New NotImplementedException("This hardware parser is not implemented yet!")
    End Function

    Public Overridable Function ParseAll(InxiToken As JToken, SystemProfilerToken As NSArray) As Dictionary(Of String, HardwareBase) Implements IHardwareParser.ParseAll
        Throw New NotImplementedException("This hardware parser is not implemented yet!")
    End Function

    ''' <summary>
    ''' The base Linux hardware parser
    ''' </summary>
    Public Overridable Function ParseLinux(InxiToken As JToken) As HardwareBase
        Throw New NotImplementedException("This hardware parser is not implemented yet on Linux!")
    End Function

    ''' <summary>
    ''' The base macOS hardware parser
    ''' </summary>
    Public Overridable Function ParseMacOS(SystemProfilerToken As NSArray) As HardwareBase
        Throw New NotImplementedException("This hardware parser is not implemented yet on macOS!")
    End Function

    ''' <summary>
    ''' The base Windows hardware parser
    ''' </summary>
    Public Overridable Function ParseWindows(WMISearcher As ManagementObjectSearcher) As HardwareBase
        Throw New NotImplementedException("This hardware parser is not implemented yet on Windows!")
    End Function

    ''' <summary>
    ''' The base Linux hardware parser
    ''' </summary>
    Public Overridable Function ParseAllLinux(InxiToken As JToken) As Dictionary(Of String, HardwareBase)
        Throw New NotImplementedException("This hardware parser is not implemented yet on Linux!")
    End Function

    ''' <summary>
    ''' The base macOS hardware parser
    ''' </summary>
    Public Overridable Function ParseAllMacOS(SystemProfilerToken As NSArray) As Dictionary(Of String, HardwareBase)
        Throw New NotImplementedException("This hardware parser is not implemented yet on macOS!")
    End Function

    ''' <summary>
    ''' The base Windows hardware parser
    ''' </summary>
    Public Overridable Function ParseAllWindows(WMISearcher As ManagementObjectSearcher) As Dictionary(Of String, HardwareBase)
        Throw New NotImplementedException("This hardware parser is not implemented yet on Windows!")
    End Function

End Class

File: Inxi.NET/Base/HardwareParserBase.cs
using Claunia.PropertyList;
using InxiFrontend.Base;
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Management;

namespace InxiFrontend
{
    /// <summary>
    /// Base for hardware parser
    /// </summary>
    public abstract class HardwareParserBase : IHardwareParser
    {
        /// <summary>
        /// Parses a hardware
        /// </summary>
        /// <param name="InxiToken">Inxi token</param>
        /// <param name="SystemProfilerToken">system_profiler token</param>
        /// <returns><see cref="IHardware"/> containing all the information</returns>
        /// <exception cref="NotImplementedException"></exception>
        public virtual IHardware Parse(JToken InxiToken, NSArray SystemProfilerToken)
        {
            throw new NotImplementedException("This hardware parser is not implemented yet!");
        }

        /// <summary>
        /// Parses a list of hardware
        /// </summary>
        /// <param name="InxiToken">Inxi token</param>
        /// <param name="SystemProfilerToken">system_profiler token</param>
        /// <returns>A dictionary containing list of hardware</returns>
        /// <exception cref="NotImplementedException"></exception>
        public virtual Dictionary<string, IHardware> ParseAll(JToken InxiToken, NSArray SystemProfilerToken)
        {
            throw new NotImplementedException("This hardware parser is not implemented yet!");
        }

        /// <summary>
        /// Parses a list of hardware
        /// </summary>
        /// <param name="InxiToken">Inxi token</param>
        /// <param name="SystemProfilerToken">system_profiler token</param>
        /// <returns>A list containing list of hardware</returns>
        /// <exception cref="NotImplementedException"></exception>
        public virtual List<IHardware> ParseAllToList(JToken InxiToken, NSArray SystemProfilerToken)
        {
            throw new NotImplementedException("This hardware parser is not implemented yet!");
        }

        /// <summary>
        /// The base Linux hardware parser
        /// </summary>
        public virtual IHardware ParseLinux(JToken InxiToken)
        {
            throw new NotImplementedException("This hardware parser is not implemented yet on Linux!");
        }

        /// <summary>
        /// The base Windows hardware parser
        /// </summary>
        public virtual IHardware ParseWindows(ManagementObjectSearcher WMISearcher)
        {
            throw new NotImplementedException("This hardware parser is not implemented yet on Windows!");
        }

        /// <summary>
        /// The base Linux hardware parser
        /// </summary>
        public virtual Dictionary<string, IHardware> ParseAllLinux(JToken InxiToken)
        {
            throw new NotImplementedException("This hardware parser is not implemented yet on Linux!");
        }

        /// <summary>
        /// The base Windows hardware parser
        /// </summary>
        public virtual Dictionary<string, IHardware> ParseAllWindows(ManagementObjectSearcher WMISearcher)
        {
            throw new NotImplementedException("This hardware parser is not implemented yet on Windows!");
        }

        /// <summary>
        /// The base Linux hardware parser
        /// </summary>
        public virtual List<IHardware> ParseAllToListLinux(JToken InxiToken)
        {
            throw new NotImplementedException("This hardware parser is not implemented yet on Linux!");
        }

        /// <summary>
        /// The base Windows hardware parser
        /// </summary>
        public virtual List<IHardware> ParseAllToListWindows(ManagementObjectSearcher WMISearcher)
        {
            throw new NotImplementedException("This hardware parser is not implemented yet on Windows!");
        }

    }
}